python/ray/data/tests/test_backpressure_policies.py

```diff
@@ -297,6 +297,30 @@ def test_no_deadlock(self):
             consumer_timestamps,
         )
 
+    def test_no_deadlock_for_resource_contention(self):
+        """Test no deadlock in case of resource contention from
+        non-Data code."""
+        # Create a non-Data actor that uses 4 CPUs, only 1 CPU
+        # is left for Data. Currently Data StreamExecturo still
+        # incorrectly assumes it has all the 5 CPUs.
+        # Check that we don't deadlock in this case.
+
+        @ray.remote(num_cpus=4)
+        class DummyActor:
+            def foo(self):
+                return None
+
+        dummy_actor = DummyActor.remote()
```

Comment:
Right now, StreamExecutor is using the total cluster resources as the default global limits, not the available resources. This is because the global limits are updated periodically during the execution to accommodate auto-scaling. And we cannot use available resources because we don't know how many of them are used by Data. 
So it doesn't matter when the actor is create.

Reply:
Got it, thanks.